xlive/Blam/Engine/items/weapon_definitions.h

```diff
@@ -0,0 +1,519 @@
+    real32 dual_weapon_error_deceleration_time_seconds;   // the continuous idle time it takes for the weapon to return to its initial error
+    real32 runtime_dual_weapon_error_acceleration_rate;
+    real32 runtime_dual_weapon_error_deceleration_rate;
+    real32 dual_weapon_error_minimum_error_degrees;
+    real_bounds dual_weapon_error_angle_degrees;
+    real32 dual_wield_damage_scale;
+
+    // Explaination("projectile", "")
+
+    e_barrel_distribution_function distribution_function;
+    int16 projectiles_per_shot;
+    real_angle distribution_angle_degrees;
+    real_angle projectile_minimum_error_degrees;
+    real_bounds projectile_error_angle_degrees;
+    real_point3d first_person_offset_world_units;   // +x is forward, +z is up, +y is left
+    e_damage_reporting_type damage_effect_reporting_type;
+    int8 pad1[3];
+    tag_reference projectile;       // proj
+    tag_reference damage_effect;    // jpt!
+
+    // Explaination("misc", "")
+
+    real32 ejection_port_recovery_time; // the amount of time (in seconds) it takes for the ejection port to transition from 1.0 (open) to 0.0 (closed) after a shot has been fired
+    real32 illumination_recovery_time;  // the amount of time (in seconds) it takes the illumination function to transition from 1.0 (bright) to 0.0 (dark) after a shot has been fired
+    real32 heat_generated_per_round;    // the amount of heat generated each time the trigger is fired
+    real32 age_generated_per_round;     // the amount the weapon ages each time the trigger is fired
+    real32 overload_time_seconds;       // the next trigger fires this often while holding down this trigger
+    
+    // Explaination("angle change (recoil)", "")
+
+    real_bounds angle_change_per_shot;  // angle change per shot of the weapon during firing
+    real32 angle_change_acceleration_time_seconds;   // the continuous firing time it takes for the weapon to achieve its final angle change per shot
+    real32 angle_change_deceleration_time_seconds;   // the continuous idle time it takes for the weapon to return to its initial angle change per shot
+
+    /// function used to scale between initial and final angle change per shot
+    e_transition_function_type angle_change_function;
+    int16 pad;
+    real32 runtime_angle_change_acceleration_rate;
+    real32 runtime_angle_change_deceleration_rate;
+    real32 runtime_illumination_recovery_rate;
+    real32 runtime_ejection_port_recovery_rate;
+    real32 runtime_rate_of_fire_acceleration_rate;
+    real32 runtime_rate_of_fire_deceleration_rate;
+    real32 runtime_error_acceleration_rate;
+    real32 runtime_error_deceleration_rate;
+    tag_block<barrel_firing_effect> firing_effects; // firing effects determine what happens when this trigger is fired
+};
+TAG_BLOCK_SIZE_ASSERT(weapon_barrel_definition, 236);
+
+struct _weapon_definition
```

Comment:
maybe we should take some liberties in renaming things like this so it's easier to figure out what exactly we are looking at? idk just a thought I know we have _device_definition too now.

Reply:
most of the definitions go by that naming scheme so there shouldn't be much confusion. See _item_definition